src/metabase/app_db/custom_migrations.clj

```diff
@@ -1823,3 +1823,13 @@
 
 (define-migration MoveExistingAtSymbolUserAttributes
   (reserve-at-symbol-user-attributes/migrate!))
+
+(define-migration BackfillTransformTargetDbId
+  (let [update! (fn [{:keys [id target]}]
+                  (let [target-map (json/decode target)
+                        db-id      (get target-map "database")]
+                    (when db-id
```

Comment:
If we can't determine the database id, then the next migration, which makes the field non-null, will fail.

I wonder if we should rather fail within this transaction, because then we can at least dump the `id`, `source`, and `target` of the offending transform.